gateway-server/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptorHandler.java

```diff
@@ -84,6 +85,8 @@
 
     private static Map<String, ServiceDiscovery> discoveryInstances = new HashMap<>();
 
+    private static final Set<String> ALLOWED_SERVICES_WITHOUT_URLS_AND_PARAMS = Collections.singleton("KNOX");
```

Comment:
I'm not sure I like this hardcoded, but overall change looks good.

Reply:
I don't have any better ideas right now. Could technically be a config, but that seems overkill to so eh.